pandas/tests/scalar/timedelta/test_timedelta.py

```diff
@@ -258,7 +258,9 @@ def test_unit_parser(self, units, np_unit, wrapper):
             if unit == "M":
                 expected = Timedelta(np.timedelta64(2, "m").astype("timedelta64[ns]"))
 
-            result = to_timedelta(f"2{unit}")
+            warning = None if unit != "m" else FutureWarning
```

Comment:
change this to parameterize this (e.g. elminate the for loop and parameterize on units)

then add a parameter whether should test for FutureWarning or not (could be bool is ok).

then in the body you can easily check this for both `to_timedelta` and `Timedelta`

Reply:
We have approx 50 values that have no warning and just one ("m") that has a warning. IMO a parameter for this would be overkill. But if we want this then I can change it.

Reply:
Updated the test. IMO `test_unit_parser` could refactored even further: split in 2 tests: `test_unit_parser_array` and `test_unit_parser_scalar` and apply a parameter on the function that converts its to timedelta. Do we want to do this?